File: backend/src/main/java/com/storecontrol/backend/models/Association.java
package com.storecontrol.backend.models;

import com.storecontrol.backend.controllers.request.association.RequestAssociation;
import com.storecontrol.backend.controllers.request.association.RequestUpdateAssociation;
import jakarta.persistence.*;
import jakarta.validation.Valid;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.List;
import java.util.UUID;

@Entity
@Table(name = "associations")
@Getter
@NoArgsConstructor
public class Association {
    @Id @GeneratedValue(generator = "UUID")
    private UUID uuid;
    @Column(name = "association_name")
    private String associationName;
    @Column(name = "principal_name")
    private String principalName;
    @OneToMany(mappedBy = "association")
    private List<Stand> stands;
    private Boolean valid;

    public Association(@Valid RequestAssociation request) {
        this.associationName = request.associationName();
        this.principalName = request.principalName();
        this.valid = true;
    }

    public void updateAssociation(RequestUpdateAssociation request) {
        if (request.associationName() != null) {
            this.associationName = request.associationName();
        }
        if (request.principalName() != null) {
            this.principalName = request.principalName();
        }
    }

    public void deleteAssociation() {
        this.valid = false;
    }
}

File: frontend/src/data/stands/Association.ts
export default interface Association {
  uuid: string;
  associationName: string;
  principalName: string;
  associationKey: string;
}

export interface SummaryAssociation {
  uuid: string;
  associationName: string;
  associationKey: string;
}

export interface CreateAssociation {
  associationName: string;
  principalName: string;
  associationKey: string;
}

export interface UpdateAssociation {
  uuid: string;
  associationName?: string;
  principalName?: string;
  associationKey?: string;
}
